components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1354,7 +1383,15 @@ where
                 }
                 CmdType::IngestSst => {
                     assert!(ctx.kv_wb.is_empty());
-                    self.handle_ingest_sst(&ctx.importer, &ctx.engine, req, &mut ssts)
+                    assert_eq!(requests.len(), 1);
+                    if self.pending_sst.is_none() {
+                        let sst = req.get_ingest_sst().get_sst();
+                        self.handle_ingest(&ctx.importer, sst)?;
+                        ctx.pending_ssts.push(sst.to_owned());
+                        return Ok((RaftCmdResponse::default(), ApplyResult::Yield));
```

Comment:
It is not necessary to make this fsm yield.....  In fact, we do not hope the fsm yield when there is no pending sst.
If there has been a pending sst, we only need to call `ctx.commit(self)` at once.

Reply:
There are two `pending_sst` here. 😂

`ApplyContext.pending_ssts` is the real pending ssts. These ssts will be ingested in `write_to_db`.
`ApplyDelegate.pending_sst` just saved a `SSTMetaInfo` for responsing ingest request after recovering from yield. (Maybe we can find a better name)
But I actually found a bug here. In `handle_raft_entry_normal` , `ctx.commit(self)` is called every time because of the ingest request. `ApplyContext.pending_ssts` will always be empty here.

Do you have any idea that we can avoid yield when first applying a ingest request?